### PR TITLE
(PUP-7521) Fix further FQDN Test Errors

### PR DESCRIPTION
**Pull request by jcoconnor**
This is a follow-on from #6031 to apply the same changes to
- acceptance/tests/catalog_with_binary_data.rb &
- acceptance/tests/direct_puppet/supports_utf8.rb
Otherwise these tests fail with a similar error.

### CODE REVIEW

**Comment by joshcooper** [comment]
I think we want to use `Host#node_name` rather than the `fqdn` fact. By default, an agent's identity is its `fqdn`, but if a `certname` is specified or the `node_name_fact` is specified, then the agent's identity will be different, which has the side-effect of changing the path to the agent's report on the master, e.g. `$reportsdir/<node_name>.yaml`. For example, see PUP-6017, PUP-5432.

**Comment by jcoconnor** [comment]
@joshcooper Will wait until @ericldelaney returns as can discuss further w.r.t. this applying to the other 4 tests that are impacted by this.

File: acceptance/tests/direct_puppet/supports_utf8.rb [hunk 1]
```diff
@@ -50,18 +53,18 @@
   step 'run agent(s)' do
     with_puppet_running_on(master, {}) do
       agents.each do |agent|
-        fqdn = agent.hostname
+        fqdn = agent_to_fqdn(agent)
         config_version = ''
         config_version_matcher = /configuration version '(\d+)'/
-        on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname}"),
+        on(agent, puppet("agent -t --environment '#{tmp_environment} --server #{master.hostname}'"),
```
Comment (smortex) [inline acceptance/tests/direct_puppet/supports_utf8.rb]: Second quote position look strange…

Comment (ericldelaney) [inline acceptance/tests/direct_puppet/supports_utf8.rb]: yeah, the single quote should just be on the tmp environment path:  '#{tmp_environment}'

File: acceptance/tests/direct_puppet/supports_utf8.rb [hunk 2]
```diff
@@ -50,18 +53,18 @@
...
           config_version = agent_out.match(config_version_matcher)
         end
         on(agent, "cat #{tmp_file[fqdn]}").stdout do |result|
           assert_equal(file_contents, result, 'file contents did not match accepted')
         end
         on(agent, "rm #{tmp_file[fqdn]}", :accept_all_exit_codes => true)
-        on(agent, puppet("agent -t --environment #{tmp_environment} --server #{master.hostname} --use_cached_catalog"),
+        on(agent, puppet("agent -t --environment '#{tmp_environment}' --server #{master.hostname} --use_cached_catalog"),
            :acceptable_exit_codes => 2).stdout do |agent_out|
           assert_match(config_version_matcher, result, 'agent did not use cached catalog')
```
Comment (ericldelaney) [inline acceptance/tests/direct_puppet/supports_utf8.rb]: I know that you didn't change this, but where is "result" for this assert_match coming from?

Should it be "agent_out" instead?

**Comment by jcoconnor** [comment]
@joshcooper and @ericldelaney I have revised this PR to only fixe the ```acceptance/tests/catalog_with_binary_data.rb``` as this was the only test actually failing in the Imaging Acceptance tests.

@ericldelaney has pointed out that this is a new file that was added by @johnduarte after he had made the FQDN fix to the other tests.

**Comment by ericldelaney** [comment]
👍

**Comment by johnduarte** [comment]
@jcoconnor is this PR still blocked?

**Comment by jcoconnor** [comment]
@johnduarte oops - thought I had removed block. Done now

**Pull request merged**